regexec.c

```diff
@@ -1884,13 +1884,12 @@ STMT_START {
         previous_occurrence_end = s;                        \
     }
 
-/* This differs from the above macros in that it is passed a single byte that
- * is known to begin the next occurrence of the thing being looked for in 's'.
- * It does a memchr to find the next occurrence of 'byte', before trying 'COND'
- * at that position. */
-#define REXEC_FBC_FIND_NEXT_UTF8_BYTE_SCAN(byte, COND)      \
+/* This is like the above macro except the function returns NULL if there is no
+ * occurrence, and there is a further condition that must be matched besides
+ * the function */
+#define REXEC_FBC_FIND_NEXT_UTF8_SCAN_COND(f, COND)         \
     while (s < strend) {                                    \
-        s = (char *) memchr(s, byte, strend -s);            \
+        s = (char *) f;                                     \
         if (s == NULL) {                                    \
```

Comment:
I looked at this and found it a bit confusing. In the macros above, `f` is a function that is called, and it's written as `(f)`.

Here it's also (effectively) a function?

And shouldn't it also be written as `(f)`, partly to be consistent, partly because, dammit, macros, it thought that it is usually seen as good defensive practice to assume that what comes in might be an expression which if just interpolated as text, can go wrong thanks to operator precedence?

Reply:
It should be (f).  I'm adding comments to try to address the confusion